pkgs/development/libraries/freeimage/default.nix

```diff
@@ -1,37 +1,23 @@
-{ lib, stdenv, fetchurl, unzip, darwin }:
-
-# TODO: consider unvendoring various dependencies (libpng, libjpeg,
-# libwebp, zlib, ...)
+{ lib, stdenv, fetchsvn, unzip, darwin, libtiff
+, libpng, zlib, libwebp, libraw, openexr, openjpeg
+, libjpeg, jxrlib, pkgconfig, fetchpatch }:
 
+# This has been partially adapted from https://src.fedoraproject.org/rpms/freeimage
 stdenv.mkDerivation {
-  name = "freeimage-3.18.0";
+  pname = "freeimage";
+  version = "unstable-r1859-3.19.0";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/freeimage/FreeImage3180.zip";
-    sha256 = "1z9qwi9mlq69d5jipr3v2jika2g0kszqdzilggm99nls5xl7j4zl";
+  src = fetchsvn {
```

Comment:
This does not work for me: 

```
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/LibOpenJPEG.2017.vcxproj.filters
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/NEWS
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/THANKS
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/bio.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/bio.c
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/cidx_manager.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/cio.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/dwt.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/cio.c
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/event.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/function_list.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/event.c
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/invert.c
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/image.h
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/image.c
A    /nix/store/bdypzn7aabqla5k593vj974amz2d43ra-svn-rr1859/FreeImage/tags/3.18.0/Source/LibOpenJPEG/invert.h
svn: E120106: ra_serf: The server sent a truncated HTTP response body.
```

Does the svn server also also has an archive option? That might work better.

Reply:
Maybe something like this with `fetchzip` is less flaky?

```
https://sourceforge.net/code-snapshots/svn/f/fr/freeimage/svn/freeimage-svn-r1859.zip
```

Reply:
This does seem to work, though I haven't finished building yet. Thanks.

Reply:
freeimage builds fine, but imv is still building, so I don't know if imv works. I changed that and rebased.

Reply:
It works perfectly.